### ISSUE TITLE
SO_LN.ADDON_WR_DT needs to be set when adding lines at a later date

### ISSUE THREAD

**Issue by kbower** [kbower/tickettest1#498]
_Imported from trac ticket 498.
Created by **kbower**
Opened in trac: Dec 14, 2010, 9:30:06 AM
Last modified in trac: Jan 5, 2011, 5:06:10 PM_
***
We need to check with Wayne about when this is required.

_Morning Wayne,

Under what circumstances must we set SO_LN.ADDON_WR_DT? After the site's books are closed or anytime after the order was committed or always.
Thanks,
Kent_

**From Wayne:**
_Kent,

Good morning, anytime after an open order is committed and a line is
added to it.

Thanks,

Wayne Liburd_

**Comment by kbower** [comment]
_**kbower** trac comment on Dec 14, 2010, 10:44:03 AM:_

This may also be useful for discount expiry dates... orderdate or orderdetail.addeddate?

At least for Kane's, we should go by the orderdate, not line date, as indicated here by Daniel.  That may need to be a system parameter for El Dorado?


```
Yes, provided nothing has been delivered.

Daniel S. Klein
Executive Vice President
Kane Furniture Corporation
Office: 727-545-9555 ext 202
Cell Phone: 727-710-3417
Fax: 727-541-6960
 

-----Original Message-----
From: Kent Bower [mailto:kb@retailarchitects.com] 
Sent: Tuesday, December 14, 2010 10:17 AM
To: Wayne Liburd; Tim Finley; Daniel Klein
Subject: EXP_DT for discounts

Daniel/Wayne/Tim,
     Not sure who is the best to ask for this business type question: 
suppose an order is written with a discount/coupon such as 10% off 
entire order or 20% off a single item.   A day or two later (after the 
discount has expired and is no longer being run), the customer calls 
back and adds an expensive line item.  Would this line item qualify for 
the discount since the discount was still running when they placed the 
original order?

Thanks,
Kent

```

**Comment by kbower** [comment]
_**kbower** trac comment on Dec 14, 2010, 3:30:08 PM:_

Asked El Dorado if they would answer the same way:
```
Yes. Our answer would be the same as Daniel's.

Thanks,
Jesús R Capó, VP, CIO 
```

**Comment by kbower** [comment]
_**rneville** trac comment on Jan 5, 2011, 9:19:31 AM:_

Replying to [kbower](comment:3):

Please see [ticket:526](526) where they have spelled out this rule

> This may also be useful for discount expiry dates... orderdate or orderdetail.addeddate?
> 
> At least for Kane's, we should go by the orderdate, not line date, as indicated here by Daniel.  That may need to be a system parameter for El Dorado?
> 
> 
> {{{
> Yes, provided nothing has been delivered.
> 
> Daniel S. Klein
> Executive Vice President
> Kane Furniture Corporation
> Office: 727-545-9555 ext 202
> Cell Phone: 727-710-3417
> Fax: 727-541-6960
>  
> 
> -----Original Message-----
> From: Kent Bower [mailto:kb@retailarchitects.com] 
> Sent: Tuesday, December 14, 2010 10:17 AM
> To: Wayne Liburd; Tim Finley; Daniel Klein
> Subject: EXP_DT for discounts
> 
> Daniel/Wayne/Tim,
>      Not sure who is the best to ask for this business type question: 
> suppose an order is written with a discount/coupon such as 10% off 
> entire order or 20% off a single item.   A day or two later (after the 
> discount has expired and is no longer being run), the customer calls 
> back and adds an expensive line item.  Would this line item qualify for 
> the discount since the discount was still running when they placed the 
> original order?
> 
> Thanks,
> Kent
> 
> }}}
> 
>

**Comment by kbower** [comment]
_**rneville** trac comment on Jan 5, 2011, 9:19:53 AM:_

Replying to [kbower](comment:4):
see [ticket:526](526)
> Asked El Dorado if they would answer the same way:
> {{{
> Yes. Our answer would be the same as Daniel's.
> 
> Thanks,
> Jesús R Capó, VP, CIO 
> }}}

**Comment by kbower** [comment]
_**rneville** trac comment on Jan 5, 2011, 9:44:22 AM:_

(In [[1014]](../commit/eb60f9ad658d3a59c7697fdce06f4bbaf98a5880))
closes #498 and fixes a bug with ArTran.transactiondefaults()

**Comment by kbower** [comment]
_**rneville** trac status change on Jan 5, 2011, 9:44:22 AM:_ **closed**

**Comment by kbower** [comment]
_**kbower** trac comment on Jan 5, 2011, 4:52:26 PM:_

Wayne,
     Suppose line 1 is written for qty 4 chairs when the order is 
written.  Suppose the order is being filled the next day and only 3 
chairs can be filled, so GERS breaks the line 1 qty => 3 and creates a 
line 2 with qty 1.
     Will line 2 (system generated because line was broken from 1) have 
a date for SO_LN.ADDON_WR_DT or will it be null?
Thanks,
Kent
----
Kent,

Good afternoon, yes GERS will split the line and SO_LN.ADDON_WR_DT will
be null.

Thanks,
 
Wayne Liburd

**Comment by kbower** [comment]
_**kbower** trac status change on Jan 5, 2011, 4:52:26 PM:_ **reopened**

**Comment by kbower** [comment]
_**kbower** trac comment on Jan 5, 2011, 5:06:10 PM:_

(In [[1018]](../commit/3640e13b267adf26ece85e3a896aa3aa721d47d6))
 * Fix orderdetails.addeddate so only set when really adding a new line (not clone break of another line).  Fixes #498
 * Fix salesperson/percentage screens with new restricted_attributes framework. see #437
 * Only write comment about adding line when it is really a new line (not clone break of another line)

**Comment by kbower** [comment]
_**kbower** trac status change on Jan 5, 2011, 5:06:10 PM:_ **closed**
